MastodonSDK/Sources/MastodonUI/View/Content/StatusView+ViewModel.swift

```diff
@@ -689,21 +685,44 @@ extension StatusView.ViewModel {
                 let instance = authentication.instance(in: context.managedObjectContext)
                 let isTranslationEnabled = instance?.isTranslationEnabled ?? false
 
-                let menuContext = StatusAuthorView.AuthorMenuContext(
-                    name: name,
-                    isMuting: isMuting,
-                    isBlocking: isBlocking,
-                    isMyself: isMyself,
-                    isBookmarking: isBookmark,
-                    isFollowed: isFollowed,
-                    isTranslationEnabled: isTranslationEnabled,
-                    isTranslated: translatedFromLanguage != nil,
-                    statusLanguage: language
-                )
-                
-                let (menu, actions) = authorView.setupAuthorMenu(menuContext: menuContext)
-                authorView.menuButton.menu = menu
-                authorView.authorActions = actions
+                authorView.menuButton.menu = UIMenu(children: [
+                    UIDeferredMenuElement({ menuElement in
+                        
+                        let domain = authContext.mastodonAuthenticationBox.domain
+
+                        Task { @MainActor in
+                            if let relationship = try? await Mastodon.API.Account.relationships(
+                                session: .shared,
+                                domain: domain,
+                                query: .init(ids: [authorId]),
+                                authorization: authContext.mastodonAuthenticationBox.userAuthorization
+                            ).singleOutput().value {
+                                guard let rel = relationship.first else { return }
```

Comment:
Showing the menu only when the relationship was successfully downloaded implies that a user can't share a status without a connection. That's fine with me.

Also, although the default `timeoutInterval` is 60 seconds, the loading-indicator stays visible ... forever, it seems? oO

Reply:
True, I've added a fallback for only share, in case the request fails or times out: https://github.com/mastodon/mastodon-ios/pull/1213/files#diff-a5af8f364c73fb6a18c8c31d7f403f9b15898cd7034f69dd8a775fa9b5461e31R721